src/platform/packages/shared/kbn-grok-ui/components/expression.tsx

```diff
@@ -33,25 +34,46 @@ export const Expression = ({
   const expression = useObservable(draftGrokExpression.getExpression$());
 
   const grokEditorRef = useRef<monaco.editor.IStandaloneCodeEditor | null>(null);
+  const { containerRef, setupResizeChecker, destroyResizeChecker } = useResizeChecker();
 
-  const onGrokEditorMount: CodeEditorProps['editorDidMount'] = (editor) => {
-    grokEditorRef.current = editor;
-  };
+  const onGrokEditorMount: CodeEditorProps['editorDidMount'] = useCallback(
+    (editor: monaco.editor.IStandaloneCodeEditor) => {
+      grokEditorRef.current = editor;
+      setupResizeChecker(editor);
+    },
+    [setupResizeChecker]
+  );
+
+  const onGrokEditorWillUnmount: CodeEditorProps['editorWillUnmount'] = useCallback(() => {
+    destroyResizeChecker();
```

Comment:
One last thing, if you could memoize those event handlers at the hook level it would be great!